### ISSUE TITLE
Validate CSV file prior to Import

### ISSUE THREAD

**Issue by mark-dce** [UCLALibrary/californica#481]
**STORY**
As a user, I would like to be able to validate my CSV prior to running my ingest job.

**ACCEPTANCE**
When I submit a CSV file for import (at the command line)
- [ ]  The system checks the CSV for Warnings and Errors
- [ ] For now the import is aborted if there are errors or warnings

**DETAILS**
*ERRORS*
* Missing required column (e.g. Title, file)

*WARNINGS*
* Missing required value in a row (file)

**Comment by val99erie** [comment]
@emcaulay In the `dlcs_ladnn_20190318_manual_edit.csv` I see what I think are duplicate columns.  There are columns for both `Title` and `Item Title`.  The one that the importer is using is `Title`.  It looks like the values are the same for those two columns, so I suspect that they are duplicates, but I wanted to check with you first to make sure that `Item Title` column isn't needed for something.

Is it ok if I write the validator to reject that `Item Title` column?

**Comment by mark-dce** [comment]
@val99erie Yes, these are duplicate columns, they also existed in the LADNN CSV files.  I think the importer should reject the extra `Item Title` column for now per the requirement: "It raises an error if any unrecognized headings exist and tells the user what's unrecognized"  

Bess and I wrote up a ticket for more nuanced handling of this case in the future (hopefully Sprint 17) - see #505 *Warn when the import CSV has extra headers*

**Comment by val99erie** [comment]
The current state of the CSV file is documented here:
[CSV Required and Allowed Columns](https://docs.google.com/document/d/14DsXb3EuwexjuTKbelcEUoZwiTqkW1SX-FZ3tjWh-F0/edit)
